src/scanner.c

```diff
@@ -3091,8 +3110,7 @@ static Int InitKernel (
     Int                 i;
 
     TLS(Input) = TLS(InputFiles);
-    TLS(Input)--;
-    (void)OpenInput(  "*stdin*"  );
+    (void)OpenInput_Internal(  "*stdin*", 1  );
```

Comment:
I actually don't consider this change to be "more minimal" than my #942. Do you?